### DISCUSSION TITLE
Is there any gm command to add talent points?

### DISCUSSION THREAD

**Discussion by xengine-qyt** [azerothcore/azerothcore-wotlk#6097] [General]
i can't found any useful gm command for add talentpoints

modify talentpoints will reset when i logout....

**Comment by xengine-qyt** [comment]
.learn all my talents this command did not work ...

**Comment by Kitzunu** [comment]
.modify talentpoints #amount

**Reply by xengine-qyt** [reply]
.modify talentpoints:
Set free talent points for selected character or character's pet. It will be reset to default expected at next levelup/login/quest reward


it is not useful

**Comment by ArmainAP** [comment]
You can achieve this by editing the SQL database:
`USE acore_characters;`
`SELECT * FROM characters;`
`UPDATE characters SET extraBonusTalentCount = <EXTRA_TALENS_COUNT> WHERE guid = <YOUR_CHARACTER_GUID>;`

**Reply by dewmguy** [reply]
Good info @ArmainAP  thanks for sharing. I use double talent points in my world config but I’m interested in writing a eluna script that will give all players their remaining talent points when they hit 80. Do you have any advice for me on that idea?

**Reply by ArmainAP** [reply]
I am not familiar with Eluna at all. If you can run DB scripts, I would do just that as it seems the easier and built in.

If any changes in code are required, take a look over Player.cpp as the talent points are set in there.

P.S.: This won't affect pet talent points, and last time I checked there wasn't anything like extraBonusTalentCount for pets. I am not sure if this is still the case. I personally worked around this by adjusting the rate multiplier of pet talent points.

**Reply by dewmguy** [reply]
sorry for the super long delay on responding to this. eluna can execute sql commands. i'm most interested in whether applying these db modifications to available talent points is permanent and will persist through level up, login, and quest rewards. as xengine-qyt pointed out in his post above in his complaint that using the gm command is only temporary, it would be pointless otherwise.

additionally, it would seem that if these points are added while the character is logged in that upon logout the db will revert. adding the talent points while the character is logged out makes the points available to spend and they are permanent.

**Reply by ArmainAP** [reply]
Yes, these modifications persist through level up, login and quest rewards. 

I never added them while logged in. The fact that they rever on log out make it sound like a server bug. Based on the C++ code, they are extra points used in the talent point calculations.

**Reply by SuryRayn** [reply]
> You can achieve this by editing the SQL database: `USE acore_characters;` `SELECT * FROM characters;` `UPDATE characters SET extraBonusTalentCount = <EXTRA_TALENS_COUNT> WHERE guid = <YOUR_CHARACTER_GUID>;`

Hi.

I have the same problem but when I use Navicat to open the characters under score_character there is no place to put characters SET extraBonusTalentCount = <EXTRA_TALENS_COUNT> WHERE guid = <YOUR_CHARACTER_GUID>; 

would u kindly tell me how this works? I am new to this.

**Reply by ArmainAP** [reply]
Hello @SuryRayn and sorry for the delayed response.

I did not use Navicat and I am not familiar with it. What I did was to SSH into the server then open the SQL database locally and navigate via SQL commands.

**Reply by tony100399** [reply]
Hello @ArmainAP  I'm pretty bad with programming but I was able to use your code on Keira3 to successfully give my character extra talent points. Still, I have a request.

Is there any way to change the talent rate per level up for a specific character? Normally I'd edit the world settings but that could affect bots. Since your previous suggestion works for a particular character I was wondering if you knew how to change the talent rate for specific GUID's

Thank you for your time

**Reply by ArmainAP** [reply]
Hello @tony100399 

Not as far as I am aware. I ended up changing config talent point multipliers in order to handle more pet talent points in my play troughs.

**Reply by dewmguy** [reply]
@tony100399 you'd have to use lua
